src/cdn_definitions/data.yaml

```diff
@@ -251,3 +251,61 @@ repo_overrides:
       if_created_after: "2023-06-01T14:00:00Z"
       key: example
       value: true
+
+# directory_layouts is used to define DirectoryLayout objects and consist of a
+# pattern, to match against content URL, and attributes for the resulting
+# object, often present as regex groups in the pattern.
+#
+# Allowed value types; string, bool, null
+#
+# With the exception of "rhelalt", which is expected to be stored as bool,
+# a "true" value is used to signal that regex matching should be applied for the
+# corresponding group in the pattern.
+#
+# E.g.;
+#   `type: true`    ==  `type = re.match(...).group("type")`
+#   `type: false`   ==  `type = False`
+#   `type: "dist"`  ==  `type = "dist"`
+directory_layouts:
+  - name: DirectoryLayoutC
+    # "/content/<type>/<platform>/<rhui>/<variant>/<major>.<minor>/<arch>/<rest>"
+    pattern: "/(?:content|shadow)/(?P<type>aus|dist|beta|(?:retired-)?els|eus|htb|rhb|tus|e4s)/(?P<platform>rhes|rhs|cf-me)/(?P<rhui>rhui/)?(?P<variant>[^/]+)/(?P<major_version>[0-9])\\.(?P<minor_version>[0-9]+)/(?P<arch>[^/]+)/(?P<rest>.+)$"
```

Comment:
I think it would be a good idea to allow these patterns to be broken over several lines, for improved readability of the patterns (and also future changes to the patterns).

There's a few ways of doing that.  If we use "(?x)" flag in the regex (corresponds to [re.VERBOSE](https://docs.python.org/3/library/re.html#re.VERBOSE)) then it's even possible to use comments inside of the patterns. Something like this should work:

```yaml
pattern: |-
  (?x)
  /
  (?:content|shadow)/
  (?P<type>aus|dist|beta|(?:retired-)?els|eus|htb|rhb|tus|e4s)/
  (?P<platform>rhes|rhs|cf-me)/
  (?P<rhui>rhui/)?
  (?P<variant>[^/]+)/
  # NOTE: does not support major_version=10, what about RHEL 10?
  (?P<major_version>[0-9])\.(?P<minor_version>[0-9]+)/
  (?P<arch>[^/]+)/
  (?P<rest>.+)$
```